inspectit-oce-core/src/main/java/rocks/inspectit/oce/core/instrumentation/config/MethodHookConfigurationResolver.java

```diff
@@ -1,23 +1,186 @@
 package rocks.inspectit.oce.core.instrumentation.config;
 
+import com.google.common.annotations.VisibleForTesting;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Value;
+import lombok.val;
 import net.bytebuddy.description.method.MethodDescription;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Component;
 import rocks.inspectit.oce.core.instrumentation.config.model.InstrumentationRule;
 import rocks.inspectit.oce.core.instrumentation.config.model.MethodHookConfiguration;
+import rocks.inspectit.oce.core.instrumentation.config.model.ResolvedDataProviderCall;
+import rocks.inspectit.oce.core.instrumentation.dataprovider.generic.DataProviderGenerator;
 
-import java.util.Set;
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 
 @Component
 public class MethodHookConfigurationResolver {
```

Comment:
I did really my best to understand this, but you need to add more explanations in the code it self, there's so much going on here, that this is quite hard to understand.. I know that the end result will be, but how exactly was it reached is not clear..